File: typescript/resistor-color-duo/resistor-color-duo.ts
enum COLOR {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white,
}

const colorCode = (color: string): number => COLOR[color];

export const decodedValue = ([colorOne, colorTwo]: string[]): number =>
  colorCode(colorOne) * 10 + colorCode(colorTwo);

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
  ]

  def self.value(colors) 
    colors.slice(0, 2).map { |color| COLORS.index(color) }.join().to_i()
  end
end

